sycl/doc/design/IVDEPAndParallelLoopAnnotations.md

```diff
@@ -0,0 +1,195 @@
+   previously-generated `llvm.loop.parallel_access_indices` due to the mismatch
+   in the semantics of `llvm.loop.paralllel_accesses` and `ivdep`. The semantics
+   are that all accesses with access groups listed in this metadata have no
+   loop-carried memory data dependences.
+2. A new metadata, `llvm.loop.no_depends_safelen` should be generated on loops
+   that are marked with the versions of ivdep that take a safelen (i.e.,
+   `intel::ivdep(safelen)` and `intel::ivdep(array, safelen)`). It will replace our
+   `llvm.loop.parallel_access_indices_safelen` for the same reason described
+   above. As with `llvm.loop.no_depends`, the semantics indicate that all accesses
+   with access groups listed in this metadata have dependence distances of at
+   least safelen.
+3. The built-in `llvm.loop.parallel_accesses` metadata should be generated on
+   loops that are marked as parallel, or in cases where the compiler knows that
+   the loop is parallel. This metadata allows the compiler to not only ignore
+   loop-carried data dependences, but also loop-carried memory-ordering dependences.
+4. The built-in `llvm.access.group` metadata should be generated for all memory
+   operations (loads, stores, and function calls) in the body of a loop that has
+   either an ivdep attribute, or is marked as parallel (by the user or the
+   compiler). The access group identifier will be passed to the corresponding
+   loop metadata described above.
+
+## SPIR-V Extension
+
+The SPIR-V extension for this design doc is presented in [this sCART
+Review](https://github.com/intel-innersource/documentation.xpu.architecture.spirv-extension-drafts/pull/162)
+(will update link to Khronos repo when extension review is complete). This list
+summarizes the additions:
+
+1. Access groups defined using an LLVM `llvm.access.group` metadata are
+represented as a literal in SPIR-V.
+2. A new instruction `OpAGListINTEL` is used to aggregate multiple access groups
+   into a single list.
+3. A new Decoration `AccessGroupINTEL` is used to specify which access groups a
+   function call or atomic instruction belongs to.
+4. A new Memory Operand `AccessGroupINTELMask` is used to specify which access
+   groups a memory operation belongs to.
+5. Two new Loop Controls are added to represent `llvm.loop.parallel_accesses`
+   and `llvm.loop.no_depends/no_depends_safelen`. They are called
+   `ParallelAccessesINTEL` and `DependencyAccessesINTEL` accordingly.
+   1. `ParallelAccessesINTEL` takes one or more `OpAGListINTEL` access group
+      lists and specifies that for each list, all the accesses that belong to
+      the specified access groups do not have loop-carried dependences.
+   2. `DependencyAccessesINTEL` takes one or more pairs. The first element in
+      each pair is a list of access groups from `OpAGListINTEL`, and the second
+      element is an integer `S` that corresponds to the safelen. 
+
+## Translation Rules
```

Comment:
I think this needs to be updated to follow the convention of https://github.com/KhronosGroup/SPIRV-LLVM-Translator/blob/main/docs/SPIRVRepresentationInLLVM.rst.  Specifically, the current practice is for the translator not to translate to or from vendor-specific named metadata.  Also, that spec will likely need to be augmented with general rule for loop controls.